model/lifecycle.go

```diff
@@ -742,6 +742,9 @@ func createTasksForBuild(ctx context.Context, creationInfo TaskCreationInfo) (ta
 	taskMap := make(map[string]*task.Task)
 	for _, t := range tasksToCreate {
 		id := execTable.GetId(creationInfo.Build.BuildVariant, t.Name)
+		if id == "" {
```

Comment:
nit: can we move this into `createOneTask` for empty id protection against potential future uses of it?